From 9dc89991d427bab2e630f7c7d8b8ec3e10aeae71 Mon Sep 17 00:00:00 2001
From: "Langer, Thomas" <thomas.langer@intel.com>
Date: Tue, 4 Aug 2020 14:24:50 +0200
Subject: [PATCH] PONRTSYS-7673: xgmac_mdio: Do not release gpio after reset

as this will revert the pin to default again and keep the PHY in reset.

Fixes: 169a999cad7f (PONRTSYS-7673: xrx500-eth 18000000.ssx1_1:eth@a00000 eth0_0 (uninitialized): Unable to find phydev)
---
 drivers/net/ethernet/lantiq/switch-api/mac/xgmac_mdio.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/switch-api/mac/xgmac_mdio.c b/drivers/net/ethernet/lantiq/switch-api/mac/xgmac_mdio.c
index b2ffb2865402..cd1078f7c020 100644
--- a/drivers/net/ethernet/lantiq/switch-api/mac/xgmac_mdio.c
+++ b/drivers/net/ethernet/lantiq/switch-api/mac/xgmac_mdio.c
@@ -499,7 +499,7 @@ int xgmac_mdio_reset(struct mii_bus *bus)
 	if (!dev)
 		return 0;
 
-	descs = gpiod_get_array_optional(dev, "reset", GPIOD_OUT_LOW);
+	descs = devm_gpiod_get_array_optional(dev, "reset", GPIOD_OUT_LOW);
 	if (!descs)
 		return 0;
 
@@ -515,8 +515,6 @@ int xgmac_mdio_reset(struct mii_bus *bus)
 
 	msleep(100);
 
-	gpiod_put_array(descs);
-
 	return 0;
 }
 
